src/include/thread_group.h

```diff
@@ -0,0 +1,56 @@
+/*-
+ * Copyright (c) 2014-2016 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * WT_WORKER_THREAD --
+ *	Encapsulation of a utility worker thread.
+ */
+struct __wt_worker_thread {
+	WT_SESSION_IMPL *session;
+	u_int id;
+	wt_thread_t tid;
+#define	WT_WORKER_THREAD_RUN		0x01
+	uint32_t flags;
+
+	/* The runner function used by all workers. */
+	int (*run_func)(WT_SESSION_IMPL *session, WT_WORKER_THREAD *context);
```

Comment:
Per my general comment, if we can figure out a clean way to have different functions, then we'd want to get rid of the function in the thread group since different worker threads in the same group could have different functions.  I'd also prefer these all be named WT_THREAD (remove WORKER) - similar to the util comment earlier.